pkg/tpkg/manager.go

```diff
@@ -367,12 +367,33 @@ func (m *ProjectPkgManager) InstallLocalPkg(ctx context.Context, name string, pa
 		}
 		return "", m.ui.ReportError("Target '%s' is not a directory: %v", path, err)
 	}
-	if name == "" {
-		name = filepath.Base(abs)
-	}
 
-	if !isValidName(name) {
-		return "", m.ui.ReportError("Invalid name: '%s'", name)
+	if name == "" {
+		target_spec := filepath.Join(abs, "package.yaml")
+		if isFile, err := isFile(target_spec); !isFile || err != nil {
+			if err == nil {
+				if !doesPathExist(target_spec) {
+					return "", m.ui.ReportError("Missing package.yaml in '%s'", path)
+				}
+				return "", m.ui.ReportError("Package.yaml in '%s' is not a file", path)
```

Comment:
I'd probably find a way to avoid capitalizing the filename here (Package.yaml) here and below.

Reply:
Done, but not super happy with the new error message.
Will try to come up with a better one for a future PR.